src/Raven.Server/Monitoring/Snmp/Objects/Server/1.19/ServerLongestTransaction.cs

```diff
@@ -0,0 +1,80 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Lextm.SharpSnmpLib;
+using Raven.Client.Util;
+using Raven.Server.ServerWide;
+using Voron;
+
+namespace Raven.Server.Monitoring.Snmp.Objects.Server
+{
+    public class ServerLongestTransaction : ScalarObjectBase<TimeTicks>
+    {
+        private readonly ServerStore _store;
+
+        public ServerLongestTransaction(ServerStore store)
+            : base(SnmpOids.Server.ServerLongestTransaction)
+        {
+            _store = store;
+        }
+
+        protected override TimeTicks GetData()
+        {
+            var now = SystemTime.UtcNow;
+            TryGetOldestTransaction(_store._env, out var oldestTransaction);
+
+            foreach (var storageEnvironment in GetAllStorageEnvironmentsForLoadedDatabases())
+            {
+                if (TryGetOldestTransaction(storageEnvironment, out var databaseOldestTransaction) == false)
+                    continue;
+
+                if (oldestTransaction == DateTime.MinValue)
```

Comment:
Is there a possibility we would get MinValue sometimes? I guess the server does not always have a transaction open right?

Then if MinValue would be reported, monitoring would sometimes report that there's an ancient transaction running :)

I propose using DateTime.MaxValue or DateTime.Now as default.

Reply:
returning 0